autogpt_platform/frontend/src/hooks/useCredits.ts

```diff
@@ -4,10 +4,21 @@ import { useCallback, useEffect, useMemo, useState } from "react";
 import { loadStripe, Stripe } from "@stripe/stripe-js";
 import { useRouter } from "next/navigation";
 
-export default function useCredits(): {
+export default function useCredits({
+  fetchInitialCredits = false,
+  fetchInitialAutoTopUpConfig = false,
+  fetchInitialTransactionHistory = false,
+  fetchStripeLibrary = false,
+}: {
+  fetchInitialCredits?: boolean;
+  fetchInitialAutoTopUpConfig?: boolean;
+  fetchInitialTransactionHistory?: boolean;
+  fetchStripeLibrary?: boolean;
+} = {}): {
```

Comment:
I've started seeing unrequired requests being fired a couple of times due to the user of `useEffect`s on this function, I'm explicitly toggling those now to only use it when it's needed.

Reply:
I updated this to `fetchTopUpLibrary` for clarity for now.
I think the fetches were pretty clear, it's the same information as the variable that you are using.